[WIP] Prep Updates for Merge

billdybas opened the pull request:
@vuzDylan I want us to get `updates` polished and ready to be merged into `master` and then use `master` for deployments (and remove outdated branches).

TODO:
- [x] Update the README to explain how `updates` is deployed
- [x] Add the secrets needed for rit-sse/node-api#74 (`MAILGUN_DOMAIN`, `MAILGUN_SECRET`) to the `node-api` configs
- [ ] Double check all the configs for accuracy (Glancing at some, it looks like there was some copy-pasta and comments, naming, etc. don't match the intent of the config)
- [ ] Determine if there are any containers / services we no longer need
- [ ] Get Alms + Slushbot working properly

Closes #8

billdybas committed:
Add Back EOS GIF

vuzDylan committed:
Maybe works

billdybas committed:
Change HOST to its Value

Did I just fix Slushbot?

billdybas committed:
Add '_FILE' to end of ENV Vars

This is what 'file_env' is looking for

billdybas committed:
Update Deploy Scripts

billdybas committed:
Use HOST in .env & Update 'member' References to 'alumni'

billdybas committed:
Add Bootstrap Script

billdybas committed:
Update README

billdybas commented:
@vuzDylan Can you check this for accuracy and see if Alms / Slushbot now works?

billdybas committed:
Update 'sticky' to 'stickiness'

'sticky' is deprecated

billdybas committed:
Update Traefik Let's Encrypt Config

billdybas committed:
Update Stack Name & Fix Typo

7imbrook commented on README.md:
```diff
@@ -1,20 +1,35 @@
-3. From here you can start any service you'd like.
-    * SSEPROD: `HOST=sseprod.se.rit.edu ./deploy` running this will also prompt you for any secrets that aren't created. Have those on hand (details in their respective repo)
-    * allendoesphysic: `HOST=allendoesphysics.com ./deploy` is enough
+Coming Soon.
```
Really?

billdybas replied:
💩

7imbrook commented on loadbalancer/deploy:
```diff
@@ -1,4 +1,4 @@
 
-# Node because the secrets can change we may have to delete the stack to update
-docker stack deploy -c docker-compose.lb.yml lb
+# Note because the secrets can change we may have to delete the stack and recreate it to update
```
This may have changed in recent light of docker swarm changes.

billdybas replied:
I had to re-create the stack in order to update the Traefik config.

Though, I've now changed it to use "configs" instead of "secrets" which you may be right about - will have to investigate.

7imbrook commented on sse.rit.edu/deploy:
```diff
@@ -2,22 +2,24 @@
-echo "Secrets ok";
+echo "Secrets Ok";
 
-docker stack deploy --compose-file=docker-compose.database.yml db --detach=false
```
Is this default now?

7imbrook commented on README.md:
```diff
@@ -1,20 +1,35 @@
-#### Notes
-If running locally, you will want to updated the `traefik.frontend.rule=Host:...` label on the service to be able to access them from localhost. or a localhost host record.
-Setting HOST environment var will override this
+Need to back up the prod Postgres database? Run `backup_prod.sh`.
```
good add. Should get this in cron somewhere

7imbrook commented on loadbalancer/traefik.toml:
```diff
@@ -4,18 +4,23 @@ defaultEntryPoints = ["http", "https"]
-        entryPoint = "https"
-    [entryPoints.https]
-    address = ":443"
-    	[entryPoints.https.tls]
```
Why the whitespace changes?

billdybas replied:
They were 4 spaces. Everything else was 2 spaces. Consistency, man.

7imbrook commented on planyourcourses.com/docker-compose.course.yml:
```diff
@@ -10,7 +10,7 @@ services:
         - "traefik.docker.network=external"
-        - "traefik.backend.loadbalancer.sticky=true"
+        - "traefik.backend.loadbalancer.stickiness=true"
         - "traefik.backend.loadbalancer.swarm=true"
```
if our lb is swarm, the stickiness doesn't matter, the traefik will always proxy directly to the swarm lb which isn't sticky

billdybas replied:
@vuzDylan made things 'sticky' and I don't really understand what that means. I was updating the syntax since 'sticky' is deprecated.

7imbrook commented on sse.rit.edu/deploy:
```diff
@@ -2,22 +2,24 @@
         continue;
     else
-        echo "Need to set $secret";
+        echo "Need to Set $secret";
```
nit why is this capitalized?

7imbrook commented on sse.rit.edu/deploy:
```diff
@@ -2,22 +2,24 @@
 done;
 
-echo "Secrets ok";
+echo "Secrets Ok";
```
same

7imbrook commented on sse.rit.edu/docker-compose.onerepo.yml:
```diff
@@ -63,21 +61,23 @@ services:
   internal:
     external: true
-
 # All secrets are external until secrets matures a bit more and services can
```
we should follow up on this

7imbrook commented on sse.rit.edu/docker-compose.slackbot.yml:
```diff
@@ -29,8 +29,6 @@ services:
       - SLACK_CLIENT_SECRET
       - SLACK_SLASH_TOKEN
-
-
```
whitespaces changes?

7imbrook reviewed (commented):
Just leaving some feedback to make sure there's good clarity before moving forward.

billdybas committed:
Remove Conflicting Let's Encrypt Verification Config

Traefik expects exactly one of HTTP or DNS challenge

Also use updated syntax for loading config files
